text_generation/cpp/README.md

```diff
@@ -1,25 +1,27 @@
-# Causal LM
+# Text-generation w/ Large Language Models
 
-These applications showcase inference of a causal language model (LM). They don't have many configuration options to encourage the reader to explore and modify the source code. There's a Jupyter notebook which corresponds to these pipelines and discusses how to create an LLM-powered Chatbot: https://github.com/openvinotoolkit/openvino_notebooks/tree/main/notebooks/254-llm-chatbot.
+These examples showcase inference of text-generation Large Language Models (LLMs). The applications don't have many configuration options to encourage the reader to explore and modify the source code. There is also a Jupyter [notebook](https://github.com/openvinotoolkit/openvino_notebooks/tree/main/notebooks/254-llm-chatbot) which provides an example of LLM-powered Chatbot in Python.
 
 > [!NOTE]
 > This project is not for production use.
 
 ## How it works
 
-### greedy_causal_lm
+The `decoder_greedy` and `decoder_beam` applications implement text-generation pipeline with LLMs which consists of the following steps:
+ * Load a tokenizer, a detokenizer and a model in OpenVINO (`.xml` and `.bin`).
+ * A text prompt is tokenized and passed to the model.
+ * The model predicts next token in a loop until the special end of sequence (EOS) token is obtained. The predicted tokens are converted to text and printed in a streaming fashion.
 
-The program loads a tokenizer, a detokenizer and a model (`.xml` and `.bin`) to OpenVINO. A prompt is tokenized and passed to the model. The model greedily generates token by token until the special end of sequence (EOS) token is obtained. The predicted tokens are converted to chars and printed in a streaming fashion.
+ Each application demonstrates a single text-generation scenario:
```

Comment:
I thought about "specific" in the beginning but decided to use "single" since we agreed to have a separate app for one use case. BTW, the word "one" sounds also not bad.